script/product-list.js

```diff
@@ -0,0 +1,164 @@
+                    </div>
+                    <button
+                      class="bag-button button button--xxl" data-id=${product.id}
+                    >
+                      ADD TO CART
+                    </button>
+                    <p class="product-status"></p>
+                    <div class="product-hide">
+                      <div class="product-action">
+                        <div class="product-action__item">
+                          <i class="fa-solid fa-heart"></i>
+                          <p>Yêu thích</p>
+                        </div>
+                        <div class="product-action__item">
+                          <i class="fa-solid fa-code-compare"></i>
+                          <p>So sánh</p>
+                        </div>
+                        <div class="product-action__item"></div>
+                        <i class="fa-solid fa-link"></i>
+                      </div>
+                    </div>
+                  </div>
+        `;
+      const productLine = `
+                            <div class="product-line">
+                    <img
+                      class="product-line__img"
+                      src=${product.img}
+                    />
+                    <div class="product-line__details">
+                      <h4 class="heading-titary">${product.title}</h4>
+                      <p class="product-line__price">${product.price}</p>
+                      <p class="product-line__description">
+                        ${product.description}
+                      </p>
+                      <div class="product-action">
+                        <div class="product-action__item">
+                          <i class="fa-solid fa-heart"></i>
+                          <p>Yêu thích</p>
+                        </div>
+                        <div class="product-action__item">
+                          <i class="fa-solid fa-code-compare"></i>
+                          <p>So sánh</p>
+                        </div>
+                        <div class="product-action__item"></div>
+                        <i class="fa-solid fa-link"></i>
+                      </div>
+                      <button
+                        class="bag-button button button--xxl" data-id=${product.id}
+                      >
+                        ADD TO CART
+                      </button>
+                    </div>
+                  </div>
+        `;
+      productListGrid.insertAdjacentHTML("afterbegin", productGrid);
+      productListLine.insertAdjacentHTML("afterbegin", productLine);
+    });
+  }
+
+  addCartButton() {
+    const bagButton = [...document.querySelectorAll(".bag-button")];
+    buttonDOM = bagButton;
+    bagButton.forEach((button) => {
+      const productId = button.dataset.id;
+
+      const inCart = cart.find((item) => item.id === productId);
+      if (inCart) {
+        button.innerText = "IN CART";
+        button.disabled = true;
```

Comment:
```suggestion
      const productAdded = cart.find((item) => item.id === productId);
      if (productAdded) {
        button.innerText = "IN CART";
        button.disabled = true;
```
Nhưng mà added thì khi nhấn thêm nó sẽ tăng amount +1 lên chứ sao lại disable nó luôn thế